internal/rpc/rpc.go

```diff
@@ -20,7 +22,11 @@ func InitializeRpcServer(node *node.Node, rpcPort int, useNats bool) (*rpc.RpcSe
 		transport, err = nats.NewNatsTransportAsServer(rpcPort)
 	} else {
 		slog.Info("Initializing Http RPC transport...")
-		transport, err = http.NewHttpTransportAsServer(fmt.Sprint(rpcPort))
+		transport, err = httpTransport.NewHttpTransportAsServer(fmt.Sprint(rpcPort))
+		if err != nil {
+			return nil, err
+		}
+		err = blockUntilHttpServerIsReady(rpcPort)
```

Comment:
Is it a bit smelly to have the RPC server send GET requests to itself to decide when some part of it is ready? I would have thought the `blockUntilReady` helper would be more naturally used by an RPC Client?

Reply:
This a good thought. The RPC client now checks for server readiness. Take a look at https://github.com/statechannels/go-nitro/pull/1652/commits/c9078d9dfc1777c2510111b0aa95aa3f0b5ecc45.